File: 0100-SameTree/Solution.java
import java.util.*;

public class Solution {
    
    static boolean isSameTree(TreeNode p, TreeNode q) {
        if (p == null && q == null) {
            return true;
        } else if (p == null || q == null) {
            return false;
        } else {
            return p.val == q.val
                && isSameTree(p.left, q.left)
                && isSameTree(p.right, q.right);
        }
    }

    public static void main(String[] args) {
        System.out.println();
    }

}

File: 0100-SameTree/solution.py
class Solution:

    def is_same_tree(self, p: TreeNode, q: TreeNode):
        if not p and not q:
            return True
        elif not p or not q:
            return False
        else:
            return (p.val == q.val
                    and self.is_same_tree(p.left, q.left)
                    and self.is_same_tree(p.right, q.right))


if __name__ == "__main__":
    s = Solution()

